### PR TITLE
auth: fix panic using WithRoot and improve JWT coverage

### PR DESCRIPTION
**Pull request by hexfusion**
ref: https://github.com/coreos/etcd/issues/9695

### CODE REVIEW

**Comment by hexfusion** [comment]
@gyuho FYI

https://jenkins-etcd-public.prod.coreos.systems/job/etcd-ci-ppc64/4075/console
`fatal: write error: No space left on device`

**Comment by gyuho** [comment]
@hexfusion 

@mkumatag Can you clean up the disk :)

**Comment by gyuho** [comment]
@hexfusion Can we add a simple unit test? Thanks!

**Comment by hexfusion** [comment]
> @hexfusion Can we add a simple unit test? Thanks!

@gyuho yes sir will address

**Comment by hexfusion** [comment]
@gyuho test added, thanks.

File: auth/store.go [hunk 1]
```diff
@@ -1284,7 +1284,7 @@ func NewTokenProvider(
 }
 
 func (as *authStore) WithRoot(ctx context.Context) context.Context {
-	if !as.IsAuthEnabled() {
+	if tt := tokenProviderType(as); tt != "simple" || !as.IsAuthEnabled() {
```
Comment (xiang90) [inline auth/store.go]: we probably should define "simple" as a const.

**Comment by hexfusion** [comment]
Updated the check and added some testing. In general, it is very awkward how NewTokenProvider accepts opts as a string, that is fine for simple but not for jwt. I didn't address this here but this should be looked at further. I also removed the util function and tests as they were not necessary.

/cc @gyuho @joelegasse

File: auth/store.go (hunk 1 above)
Comment (hexfusion) [reply auth/store.go]: @xiang90 good idea, done.

**Comment by hexfusion** [comment]
@gyuho @joelegasse @xiang90 thank you!

**Pull request merged**